app/src/main/java/br/com/lucas/pomodoroapp/ui/listTaskScreen/ListTaskViewModel.kt

```diff
@@ -16,16 +18,18 @@ class ListTaskViewModel @Inject constructor(
     private val repository: TaskRepository
 ) : ViewModel() {
 
-    val taskList = MutableLiveData<List<Task>>()
+    val taskList = MutableLiveData<List<AdapterItem>>()
     val selectionMode = MutableLiveData<Boolean>(false)
     private val tasksSelected = ArrayList<Task>()
 
-    var previousSelection: ArrayList<Int> = arrayListOf()
-        private set
+    fun convertAdapterItemToTask(adapterItem: AdapterItem): Task {
+        return adapterItem.toTaskItem()
+    }
 
-    fun syncSelection(task: Task, isSelected: Boolean) {
-        if (isSelected) {
-            val exists = tasksSelected.any { it.uid == task.uid }
+    fun syncSelection(taskAdapterItem: AdapterItem) {
+        val task = taskAdapterItem.toTaskItem()
+        if (taskAdapterItem.isSelected) {
+            val exists = tasksSelected.any { it.uid == taskAdapterItem.uid }
             if (!exists) {
```

Comment:
Maybe here another suggestion it would be:
```kotlin
if(tasksSelected.none{it.uid == taskAdapterItem.uid} )
```

Reply:
I think the none is the opposite what we need. It will be the opposite of exists, I would have to change the val name from exists to dontExists. I think will do the same thing but inverted 🤔

Reply:
I was wrong, it will increase the readability 😬

Reply:
I renamed the val to 'notExists'. Now it's better to understand! =D